app/components/bali/link/component.rb

```diff
@@ -5,17 +5,52 @@ module Link
     class Component < ApplicationViewComponent
       attr_reader :name, :href, :type, :drawer, :modal, :options
 
-      renders_one :icon, ->(**options, &block) { tag.span(**options, &block) }
+      renders_one :icon, ->(name, **options) { Icon::Component.new(name, **options) }
+
+      # Params:
+      #  name: [String] The name of the link.
+      #  href: [String] The href of the link.
+      #  type: [Symbol] or [String] This is a class for the link:
+      #   (:primary, :secondary, :success, :danger, :warning)
+
+      #  modal: [Boolean] The link opens a modal.
+      #  drawer: [Boolean] The link opens a drawer.
+      #  active_path: [String] The path of the active page.
+      #  match: [Symbol] To check if the path is exact or cotains the path.
+      #  method: [Symbol] or [String] The method to call when the link is clicked.
+
+      # rubocop:disable Metrics/ParameterLists
+      # rubocop:disable Metrics/AbcSize
+      def initialize(href:,
+                     name: '',
+                     type: nil,
+                     modal: false,
+                     drawer: false,
+                     active_path: nil,
+                     match: :exact,
+                     method: nil,
```

Comment:
Should we order this in alphabetical order? 🤔

Reply:
the attributes that do not have a default value have to be first, the other ones are ordered by relevance

Reply:
All of them have a default attribute

Reply:
`href` doesn't